tests/test_rois.py

```diff
@@ -1,209 +1,239 @@
 import json
-
 import sqlite3
 import pytest
 import numpy as np
-
+from scipy.sparse import coo_matrix
 import segmentation_labeling_app.rois.rois as roi_module
 
 
-@pytest.mark.parametrize(("coo_rows, coo_cols, coo_data, video_shape,"
-                          "segmentation_id, roi_id, threshold, expected_mask"), [
-    ([0, 0, 1, 1], [0, 1, 0, 1],
-     [0.75, 0.8, 0.9, 0.85], (3, 3), 1, 1, 0.7,
-     np.array([[1, 1, 0], [1, 1, 0], [0, 0, 0]])),
-    ([0, 0, 0, 1, 1, 1], [0, 1, 2, 0, 1, 2],
-     [0.75, 0.8, 0.5, 0.9, 0.85, 0.4], (3, 3), 1, 1, 0.7,
-     np.array([[1, 1, 0], [1, 1, 0], [0, 0, 0]]))
-])
-def test_binary_mask(coo_rows, coo_cols, coo_data, video_shape,
-                     segmentation_id, roi_id, threshold, expected_mask):
-    test_roi = roi_module.ROI(coo_rows, coo_cols, coo_data,
-                              video_shape, segmentation_id, roi_id)
-    generated_mask = test_roi.generate_binary_mask_from_threshold(threshold=threshold)
-    assert np.array_equal(generated_mask, expected_mask)
+@pytest.mark.parametrize(
+        ("data", "target_fraction", "expected"),
+        [
+            (np.linspace(0, 1.0, 1000), 0.95, 0.95),
+            (np.linspace(0, 1.0, 1000), 0.05, 0.05),
+            (np.array([1.0, 1.0, 5.0, 5.0]), 0.65, 10/12)
+            ])
+def test_cumulative_fraction(data, target_fraction, expected):
```

Comment:
I would consider adding the edge case of non convergent fraction, ie: what if all the data values are 0.7 and the requested fraction is 0.5.